test/classes-additional.coffee

```diff
@@ -0,0 +1,301 @@
+  a = new A('Machiato')
+  eq a.make(),  "Making a Machiato"
+
+  throwsB = """
+  class B extends A
+    #implied super
+    constructor: (@drink) ->
+      if @drink is 'Machiato'
+        @cost = 'expensive'
+    make: -> "Making an #{@cost} #{@drink}"
+  b = new B('Machiato', 'Cafe Ole', 'Americano')
+  eq b.make(),  "Making an expensive Machiato" 
+  """
+  throws -> CoffeeScript.run classA + throwsB, bare: yes
+
+  throwsC = """
+  class C extends A
+    # super with splats
+    constructor: (@params...) ->
+      super(@params[0])
+      console.log @params
+
+  c = new C('Machiato', 'Cafe Ole', 'Americano')
+  # eq c.make(),  "Making a Machiato" 
+  """
+  throws -> CoffeeScript.run classA + throwsC, bare: yes
+
+
+test "super and external constructors", ->
+  ctorA = (@drink) ->
+  class A 
+    constructor: ctorA
+    make: -> "Making a #{@drink}"
+
+  # # External constructor with super
+  # ctorB = (drink, flavor) ->
+  #   super(drink)
+  #   @flavor = flavor
+
+  # class B extends A
+  #   constructor: ctorB
+  # b = new B('Machiato')
+  # eq b.make(),  "Making a Machiato"
+
+test "super and external overrides", ->
+  # class A 
+  #   constructor: (@drink) ->
+  #   make: -> "Making a #{@drink}"
+
+  # # External method and super
+  # makeB = (@flavor) -> super.make() + " with #{@flavor}"
+
+  # class B extends A 
+  #   make: makeB
+  # b = new B('Machiato')
+  # eq b.make('caramel'),  "Making a Machiato with caramel"
+
+  # # External bound method and super
+  # makeC = (@flavor) => super.make() + " with #{@flavor}"
+
+  # class C extends A 
+  #   make: makeC
+  # e = new C('Machiato')
+  # eq e.make('caramel'),  "Making a Machiato with caramel"
+
+  # class D extends A 
+  # d = new D('Machiato')
+  # d.make = (@flavor) -> super.make() + " with #{@flavor}"
+  # eq d.make('caramel'),  "Making a Machiato with caramel"
+
+  # # class G extends A 
+  # # g = new G('Machiato')
+  # # g.make = (@flavor) => super.make() + " with #{@flavor}"
+  # # eq g.make('caramel'),  "Making a Machiato with caramel"
+  # # Bound function with @
+
+test "super in external prototype", ->
+    class A 
+      constructor: (@drink) ->
+      make: -> "Making a #{@drink}"
+
+    class B extends A 
+    B::make = (@flavor) -> super() + " with #{@flavor}"
+    b = new B('Machiato')
+    eq b.make('caramel'),  "Making a Machiato with caramel"
+
+    #  Fails, not bound
+    class C extends A 
+    C::make = (@flavor) => super() + " with #{@flavor}"
+    c = new C('Machiato')
+    ok c.make('caramel') isnt "Making a Machiato with caramel"
+
+    classA = """
+    class A 
+      constructor: (@drink) ->
+      make: -> "Making a #{@drink}"
+    """
+
+    # This throws since during compile super isn't in a known state.
+    # This incorrect syntax is technically valid, but is actually ES6 syntax.
+    # I suspect this will be a common error 
+    throwsD = """
+    class D extends A 
+    D::make = (@flavor) -> super.make() + " with #{@flavor}"
```

Comment:
Not sure we should have tests for syntax that isn't in CS. The discussion about `super.access` should happen in another read imo.